### PR TITLE
FIX missing refId on getData calls

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: if getData is called without refId, it can fail

## What I did
- ensure getData is called with RefId from state
- ensure refId is always lowercase just like storyId
- order of refs was indeterministic, I fixed it so it's the order they are defined in.

### CODE REVIEW

File: lib/router/src/utils.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export const parsePath: (path: string | undefined) => StoryData = memoize(1000)(
     };
 
     if (path) {
-      const [, viewMode, refId, storyId] = path.toLowerCase().match(splitPathRegex) || [];
+      const [, viewMode, refId, storyId] = path.match(splitPathRegex) || [];
```
Comment (tmeasday) [inline lib/router/src/utils.ts:21]: Does this mean URLs can include uppercase chars? This seems oddly inconsistent with our storyIds

Comment (ndelangen) [reply lib/router/src/utils.ts:21]: Do you recon it's better to enforce lowercasing refId's everywhere?

Comment (tmeasday) [reply lib/router/src/utils.ts]: I think so, if it's not too complicated

Comment (tmeasday) [reply lib/router/src/utils.ts]: Like we could just make it a requirement from the beginning

**Comment by ndelangen** [comment]
@tmeasday WDYT?

**Review by tmeasday — APPROVED** [review]
LGTM :+1:

**Pull request merged**